server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -478,6 +485,26 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point, fina
                             } else {
                                 throw new ElasticsearchParseException("geohash must be a string");
                             }
+                        } else if (COORDINATES.equals(field)) {
+                            subParser.nextToken();
+                            if (subParser.currentToken() == Token.START_ARRAY) {
+                                coordinates = new ArrayList<>();
+                                subParser.nextToken();
```

Comment:
I wonder if we should stricter here. We expect a two / three dimensional double array. WE should only accept that else throw an error?

Reply:
We do that on line 527. I noticed that the code here from before does a little validation during the parsing and then more validation on the results further down. I think we could do even more validation, like enforcing that you do not mix and match multiple types of data.

Reply:
I've added a lot more validation now